src/Microsoft.Health.Dicom.Core/Features/Retrieve/RetrieveResourceService.cs

```diff
@@ -113,10 +114,26 @@ public async Task<RetrieveResourceResponse> GetInstanceResourceAsync(RetrieveRes
             }
         }
 
-        private static Stream ResetDicomFileStream(Stream stream)
+        private static string GetResponseTransferSyntax(string requestedTransferSyntax, VersionedInstanceIdentifier identifier)
+        {
+            bool isOriginalTransferSyntaxRequested = DicomTransferSyntaxUids.IsOriginalTransferSyntaxRequested(requestedTransferSyntax);
+            if (isOriginalTransferSyntaxRequested)
+            {
+                return GetOriginalTransferSyntaxWithBackCompat(requestedTransferSyntax, identifier);
+            }
+            return requestedTransferSyntax;
+        }
+
+        /// <summary>
+        /// Existing dicom files(as of Feb 2022) do not have transferSyntax stored. 
```

Comment:
Since we are loading the Stream on retrieve, why not use the transfer syntax specified in the [file meta information module](https://dicom.nema.org/medical/dicom/current/output/html/part10.html#sect_7.1)? It's available in the DicomFile.FileMetaInfo property. I checked a few sample files locally.

Reply:
Not sure I understand. You have to parse the stream again to get this information. We are parsing and storing the information on the way in, to avoid parsing on the way out.